app/react/kubernetes/applications/CreateView/application-services/utils.ts

```diff
@@ -151,3 +162,57 @@ export function prependWithSlash(path?: string) {
   if (!path) return '';
   return path.startsWith('/') ? path : `/${path}`;
 }
+
+export function getServicePatchPayload(
+  oldService: ServiceFormValues,
+  newService: ServiceFormValues
+) {
+  const oldPayload = getServicePayload(oldService);
+  const newPayload = getServicePayload(newService);
+
+  const payload = compare(oldPayload, newPayload);
+  return payload;
+}
+
+function getServicePayload(service: ServiceFormValues): Service {
+  if (!service.Name || !service.Namespace) {
+    throw new Error('Service name and namespace are required');
+  }
+
+  // metadata
+  const labels: Record<string, string> = {};
+  if (service.ApplicationName) {
+    labels[appNameLabel] = service.ApplicationName;
+  }
+  if (service.ApplicationOwner) {
+    labels[appOwnerLabel] = service.ApplicationOwner;
+  }
+  if (service.StackName) {
+    labels[appStackNameLabel] = service.StackName;
+  }
+  const metadata: ObjectMeta = {
```

Comment:
why didn't we add `annotations` like in EE? is it only supported in EE?

Reply:
Yeah, only supported in EE